doc/python_api_reference_vDev.md

```diff
@@ -7659,9 +7688,18 @@ def cnot(
 # (in class stim.TableauSimulator)
 def copy(
     self,
+    *,
```

Comment:
Add an optional seed argument. Must not be specified if copy rng is specified.

Default no-arguments behavior should be new entropy for each copy. I realize this might be a breaking change (if so note it loudly), but it falls under the "fixing traps" allowance in the backwards compatibility guarantee. It's a trap for copies to all generate the same samples. That needs to be asked for.

Reply:
> FWIW, I think fresh_entropy=False should be the default. There is nothing wrong or surprising if copies behave the same way.

**EXTREME DISAGREEMENT**. The absolute worst thing we can do to a user is have them think something is generating independent samples when it is not! This can waste months of someone's time, and then require retraction of papers! It is super duper extremely very much the case that the correct default behavior is to re-seed when copying.

Incidentally, this is why sinter uses the much more expensive "spawn" subprocess style rather than "fork". Because these kinds of mistakes terrify me.

Reply:
Yeah, this makes sense. Thanks! However, what your argument says is that copying is a potentially very harmful operation for a sim, so should not be easily available in the API. Nevertheless, the actually useful operation - creating a new sim with "deterministic" portion of the state copied over from another sim and the RNG state reseeded with new entropy - is not a copy and naming it so is misleading.

Naming suggestions (not implemented): copy -> new_from, dup, bud, replicate... A newcomer to stim is less likely to form wrong preconceptions about what these do compared to "copy". These function names also allow us to rename the boolean arg from "also_copy_prng_state" to "fresh_entropy" or similar. The former name isn't great: it makes the reader worry that copying other parts of the sim's state may be contingent on a boolean argument, too, e.g. should this call to copy also set "also_copy_measurement_record"?

Better yet, we could merge `copy` into  `__init__` by changing the existing

```python
def __init__(self, *, seed: Optional[int] = None) -> None:
```

to

```python
def __init__(
    self,
    *,
    source: Optional[stim.TableauSimulator] = None,
    fresh_entropy: bool = True,
    seed: Optional[int] = None
) -> None:
```

WDYT?

Reply:
We could call it `fork` instead of `copy`. I don't like merging into the init method as it overloads that method with a bunch of very task-specific arguments.

Reply:
Yeah, `fork` is a very good name. I went ahead and renamed `also_copy_prng_state` to `fresh_entropy`, too.

Reply:
Should we retain a variant of `copy` for backwards-compatibility?